TestRaftLogQueue is flaky

```
13:01 $ make stress PKG=./storage TESTS=LogQueue
/var/folders/yy/4q8rrssd27vdgbr59w9qbffr0000gn/T/go-stress077357705
I1118 13:00:36.365470 76113 multiraft/multiraft.go:959  node 1 campaigning because initial confstate is [1]
I1118 13:00:36.366682 76113 storage/replica_command.go:1166  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC +1.000s
I1118 13:00:36.367911 76113 gossip/gossip.go:172  gossiping node descriptor node_id:1 address:<network_field:"localhost" address_field:"1" > attrs:<>
I1118 13:00:36.368542 76113 gossip/gossip.go:172  gossiping node descriptor node_id:2 address:<network_field:"localhost" address_field:"2" > attrs:<>
I1118 13:00:36.369144 76113 gossip/gossip.go:172  gossiping node descriptor node_id:3 address:<network_field:"localhost" address_field:"3" > attrs:<>
I1118 13:00:36.369954 76113 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
I1118 13:00:36.370049 76113 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
I1118 13:00:36.370103 76113 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- FAIL: TestRaftLogQueue (0.01s)
        client_raft_log_queue_test.go:85: raft log has not been truncated yet, afterTruncationIndex:11 originalIndex:11
FAIL
```

I've also seen this deadlock in `make testrace PKG=./storage TESTS=. TESTFLAGS=-v 2>&1 | less` , but that seems even weirder (note the `cacheQueue` garbling at the beginning) and could indicate a different problem:

```
=== RUN   TestRaftLogQueue
CacheError(err) {
is [1]
I1118 11:59:27.035663 68153 storage/replica_command.go:1166  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC +1.000s
I1118 11:59:27.036538 68153 gossip/gossip.go:172  gossiping node descriptor node_id:1 address:<network_field:"localhost" address_field:"1" > attrs:<>
I1118 11:59:27.037823 68153 gossip/gossip.go:172  gossiping node descriptor node_id:2 address:<network_field:"localhost" address_field:"2" > attrs:<>
panic: test timed out after 5m0s

goroutine 3298 [running]:
testing.startAlarm.func1()
        /usr/local/Cellar/go/1.5.1/libexec/src/testing/testing.go:703I1118 11:59:27.038971 68153 gossip/gossip.go:172  gossiping node descriptor node_id:3 address:<network_field:"localhost" address_field:"3" > attrs:<>
 +0x16b
created by time.goFunc
        /usr/local/Cellar/go/1.5.1/libexec/src/time/sleep.go:129 +0x6e

goroutine 1 [chan receive, 58 minutes]:
testing.RunTests(0x54bbc08, 0x587f760, 0xcf, 0xcf, 0x5149c01)
        /usr/local/Cellar/go/1.5.1/libexec/src/testing/testing.go:562 +0xafa
testing.(*M).Run(0xc820793f10, 0x5e7bc2e42e7a5ce5)
        /usr/local/Cellar/go/1.5.1/libexec/src/testing/testing.go:494 +0xe5
github.com/cockroachdb/cockroach/util/leaktest.TestMainWithLeakCheck(0xc820793f10)
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/leaktest/leaktest.go:34 +0x2f
github.com/cockroachdb/cockroach/storage_test.TestMain(0xc820793f10)
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/storage/main_test.go:35 +0x2f
main.main()
        github.com/cockroachdb/cockroach/storage/_test/_testmain.go:474 +0x20a

goroutine 17 [syscall, 58 minutes, locked to thread]:
runtime.goexit()
        /usr/local/Cellar/go/1.5.1/libexec/src/runtime/asm_amd64.s:1696 +0x1

goroutine 35 [semacquire, 58 minutes]:
sync.runtime_Semacquire(0x6264040)
        /usr/local/Cellar/go/1.5.1/libexec/src/runtime/sema.go:43 +0x26
sync.(*Mutex).Lock(0x626403c)
        /usr/local/Cellar/go/1.5.1/libexec/src/sync/mutex.go:82 +0x1ee
github.com/cockroachdb/cockroach/util/log.(*loggingT).lockAndFlushAll(0x6264000)
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1036 +0x3b
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x6264000)
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1030 +0x90
created by github.com/cockroachdb/cockroach/util/log.init.1
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:610 +0x117

goroutine 3272 [select, 58 minutes]:
github.com/cockroachdb/cockroach/storage.(*replicaScanner).waitAndProcess(0xc820184480, 0xecddea56f, 0x25a1aa5, 0x6263ba0, 0xc820078e40, 0xc8200e4d80, 0x0, 0xc80000000a)
:
I1118 11:59:27.034930 68153 multiraft/multiraft.go:959  node 1 campaigning because initial confstate is [1]
I1118 11:59:27.035663 68153 storage/replica_command.go:1166  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC +1.000s
I1118 11:59:27.036538 68153 gossip/gossip.go:172  gossiping node descriptor node_id:1 address:<network_field:"localhost" address_field:"1" > attrs:<>
I1118 11:59:27.037823 68153 gossip/gossip.go:172  gossiping node descriptor node_id:2 address:<network_field:"localhost" address_field:"2" > attrs:<>
panic: test timed out after 5m0s

goroutine 3298 [running]:
testing.startAlarm.func1()
        /usr/local/Cellar/go/1.5.1/libexec/src/testing/testing.go:703I1118 11:59:27.038971 68153 gossip/gossip.go:172  gossiping node descriptor node_id:3 address:<network_field:"localhost" address_field:"3" > attrs:<>
 +0x16b
created by time.goFunc
        /usr/local/Cellar/go/1.5.1/libexec/src/time/sleep.go:129 +0x6e

goroutine 1 [chan receive, 58 minutes]:
testing.RunTests(0x54bbc08, 0x587f760, 0xcf, 0xcf, 0x5149c01)
        /usr/local/Cellar/go/1.5.1/libexec/src/testing/testing.go:562 +0xafa
testing.(*M).Run(0xc820793f10, 0x5e7bc2e42e7a5ce5)
        /usr/local/Cellar/go/1.5.1/libexec/src/testing/testing.go:494 +0xe5
github.com/cockroachdb/cockroach/util/leaktest.TestMainWithLeakCheck(0xc820793f10)
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/leaktest/leaktest.go:34 +0x2f
github.com/cockroachdb/cockroach/storage_test.TestMain(0xc820793f10)
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/storage/main_test.go:35 +0x2f
main.main()
        github.com/cockroachdb/cockroach/storage/_test/_testmain.go:474 +0x20a

goroutine 17 [syscall, 58 minutes, locked to thread]:
runtime.goexit()
        /usr/local/Cellar/go/1.5.1/libexec/src/runtime/asm_amd64.s:1696 +0x1

goroutine 35 [semacquire, 58 minutes]:
sync.runtime_Semacquire(0x6264040)
        /usr/local/Cellar/go/1.5.1/libexec/src/runtime/sema.go:43 +0x26
sync.(*Mutex).Lock(0x626403c)
        /usr/local/Cellar/go/1.5.1/libexec/src/sync/mutex.go:82 +0x1ee
github.com/cockroachdb/cockroach/util/log.(*loggingT).lockAndFlushAll(0x6264000)
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1036 +0x3b
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x6264000)
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1030 +0x90
created by github.com/cockroachdb/cockroach/util/log.init.1
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:610 +0x117

goroutine 3272 [select, 58 minutes]:
github.com/cockroachdb/cockroach/storage.(*replicaScanner).waitAndProcess(0xc820184480, 0xecddea56f, :
is [1]
I1118 11:59:27.035663 68153 storage/replica_command.go:1166  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC +1.000s
I1118 11:59:27.036538 68153 gossip/gossip.go:172  gossiping node descriptor node_id:1 address:<network_field:"localhost" address_field:"1" > attrs:<>
I1118 11:59:27.037823 68153 gossip/gossip.go:172  gossiping node descriptor node_id:2 address:<network_field:"localhost" address_field:"2" > attrs:<>
panic: test timed out after 5m0s

goroutine 3298 [running]:
testing.startAlarm.func1()
        /usr/local/Cellar/go/1.5.1/libexec/src/testing/testing.go:703I1118 11:59:27.038971 68153 gossip/gossip.go:172  gossiping node descriptor node_id:3 address:<network_field:"localhost" address_field:"3" > attrs:<>
 +0x16b
created by time.goFunc
        /usr/local/Cellar/go/1.5.1/libexec/src/time/sleep.go:129 +0x6e

goroutine 1 [chan receive, 58 minutes]:
testing.RunTests(0x54bbc08, 0x587f760, 0xcf, 0xcf, 0x5149c01)
        /usr/local/Cellar/go/1.5.1/libexec/src/testing/testing.go:562 +0xafa
testing.(*M).Run(0xc820793f10, 0x5e7bc2e42e7a5ce5)
        /usr/local/Cellar/go/1.5.1/libexec/src/testing/testing.go:494 +0xe5
github.com/cockroachdb/cockroach/util/leaktest.TestMainWithLeakCheck(0xc820793f10)
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/leaktest/leaktest.go:34 +0x2f
github.com/cockroachdb/cockroach/storage_test.TestMain(0xc820793f10)
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/storage/main_test.go:35 +0x2f
main.main()
        github.com/cockroachdb/cockroach/storage/_test/_testmain.go:474 +0x20a

goroutine 17 [syscall, 58 minutes, locked to thread]:
runtime.goexit()
        /usr/local/Cellar/go/1.5.1/libexec/src/runtime/asm_amd64.s:1696 +0x1

goroutine 35 [semacquire, 58 minutes]:
sync.runtime_Semacquire(0x6264040)
        /usr/local/Cellar/go/1.5.1/libexec/src/runtime/sema.go:43 +0x26
sync.(*Mutex).Lock(0x626403c)
        /usr/local/Cellar/go/1.5.1/libexec/src/sync/mutex.go:82 +0x1ee
github.com/cockroachdb/cockroach/util/log.(*loggingT).lockAndFlushAll(0x6264000)
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1036 +0x3b
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x6264000)
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1030 +0x90
created by github.com/cockroachdb/cockroach/util/log.init.1
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:610 +0x117

goroutine 3272 [select, 58 minutes]:
github.com/cockroachdb/cockroach/storage.(*replicaScanner).waitAndProcess(0xc820184480, 0xecddea56f, 0x25a1aa5, 0x6263ba0, 0xc820078e40, 0xc8200e4d80, 0x0, 0xc80000000a)
:
I1118 11:59:27.034930 68153 multiraft/multiraft.go:959  node 1 campaigning because initial confstate is [1]
I1118 11:59:27.035663 68153 storage/replica_command.go:1166  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC +1.000s
I1118 11:59:27.036538 68153 gossip/gossip.go:172  gossiping node descriptor node_id:1 address:<network_field:"localhost" address_field:"1" > attrs:<>
I1118 11:59:27.037823 68153 gossip/gossip.go:172  gossiping node descriptor node_id:2 address:<network_field:"localhost" address_field:"2" > attrs:<>
panic: test timed out after 5m0s

goroutine 3298 [running]:
testing.startAlarm.func1()
        /usr/local/Cellar/go/1.5.1/libexec/src/testing/testing.go:703I1118 11:59:27.038971 68153 gossip/gossip.go:172  gossiping node descriptor node_id:3 address:<network_field:"localhost" address_field:"3" > attrs:<>
 +0x16b
created by time.goFunc
        /usr/local/Cellar/go/1.5.1/libexec/src/time/sleep.go:129 +0x6e

goroutine 1 [chan receive, 58 minutes]:
testing.RunTests(0x54bbc08, 0x587f760, 0xcf, 0xcf, 0x5149c01)
        /usr/local/Cellar/go/1.5.1/libexec/src/testing/testing.go:562 +0xafa
testing.(*M).Run(0xc820793f10, 0x5e7bc2e42e7a5ce5)
        /usr/local/Cellar/go/1.5.1/libexec/src/testing/testing.go:494 +0xe5
github.com/cockroachdb/cockroach/util/leaktest.TestMainWithLeakCheck(0xc820793f10)
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/leaktest/leaktest.go:34 +0x2f
github.com/cockroachdb/cockroach/storage_test.TestMain(0xc820793f10)
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/storage/main_test.go:35 +0x2f
main.main()
        github.com/cockroachdb/cockroach/storage/_test/_testmain.go:474 +0x20a

goroutine 17 [syscall, 58 minutes, locked to thread]:
runtime.goexit()
        /usr/local/Cellar/go/1.5.1/libexec/src/runtime/asm_amd64.s:1696 +0x1

goroutine 35 [semacquire, 58 minutes]:
sync.runtime_Semacquire(0x6264040)
        /usr/local/Cellar/go/1.5.1/3 +0x26
sync.(*Mutex).Lock(0x626403c)
        /usr/local/Cellar/go/1.5.1/libexec/src/sync/mutex.go:82 +0x1ee
github.com/cockroachdb/cockroach/util/log.(*loggingT).lockAndFlushAll(0x6264000)
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1036 +0x3b
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x6264000)
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1030 +0x90
created by github.com/cockroachdb/cockroach/util/log.init.1
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:610 +0x117

goroutine 3272 [select, 58 minutes]:
github.com/cockroachdb/cockroach/storage.(*replicaScanner).waitAndProcess(0xc820184480, 0xecddea56f, :
I1118 11:59:27.034930 68153 multiraft/multiraft.go:959  node 1 campaigning because initial confstate is [1]
I1118 11:59:27.035663 68153 storage/replica_command.go:1166  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC +1.000s
I1118 11:59:27.036538 68153 gossip/gossip.go:172  gossiping node descriptor node_id:1 address:<network_field:"localhost" address_field:"1" > attrs:<>
I1118 11:59:27.037823 68153 gossip/gossip.go:172  gossiping node descriptor node_id:2 address:<network_field:"localhost" address_field:"2" > attrs:<>
panic: test timed out after 5m0s

goroutine 3298 [running]:
testing.startAlarm.func1()
        /usr/local/Cellar/go/1.5.1/libexec/src/testing/testing.go:703I1118 11:59:27.038971 68153 gossip/gossip.go:172  gossiping node descriptor node_id:3 address:<network_field:"localhost" address_field:"3" > attrs:<>
 +0x16b
created by time.goFunc
        /usr/local/Cellar/go/1.5.1/libexec/src/time/sleep.go:129 +0x6e

goroutine 1 [chan receive, 58 minutes]:
testing.RunTests(0x54bbc08, 0x587f760, 0xcf, 0xcf, 0x5149c01)
        /usr/local/Cellar/go/1.5.1/libexec/src/testing/testing.go:562 +0xafa
testing.(*M).Run(0xc820793f10, 0x5e7bc2e42e7a5ce5)
        /usr/local/Cellar/go/1.5.1/libexec/src/testing/testing.go:494 +0xe5
github.com/cockroachdb/cockroach/util/leaktest.TestMainWithLeakCheck(0xc820793f10)
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/leaktest/leaktest.go:34 +0x2f
github.com/cockroachdb/cockroach/storage_test.TestMain(0xc820793f10)
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/storage/main_test.go:35 +0x2f
main.main()
        github.com/cockroachdb/cockroach/storage/_test/_testmain.go:474 +0x20a

goroutine 17 [syscall, 58 minutes, locked to thread]:
runtime.goexit()
        /usr/local/Cellar/go/1.5.1/libexec/src/runtime/asm_amd64.s:1696 +0x1

goroutine 35 [semacquire, 58 minutes]:
sync.runtime_Semacquire(0x6264040)
        /usr/local/Cellar/go/1.5.1/libexec/src/runtime/sema.go:43 +0x26
sync.(*Mutex).Lock(0x626403c)
        /usr/local/Cellar/go/1.5.1/libexec/src/sync/mutex.go:82 +0x1ee
github.com/cockroachdb/cockroach/util/log.(*loggingT).lockAndFlushAll(0x6264000)
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1036 +0x3b
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x6264000)
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:1030 +0x90
created by github.com/cockroachdb/cockroach/util/log.init.1
        /Users/tschottdorf/go/src/github.com/cockroachdb/cockroach/util/log/clog.go:610 +0x117

goroutine 3272 [select, 58 minutes]:
github.com/cockroachdb/cockroach/storage.(*replicaScanner).waitAndProcess(0xc820184480, 0xecddea56f, 0x25a1aa5, 0x6263ba0, 0xc820078e40, 0xc8200e4d80, 0x0, 0xc80000000a)
:
```

--- a/storage/client_raft_log_queue_test.go
+++ b/storage/client_raft_log_queue_test.go
@@ -24,6 +24,7 @@ import (
 
 	"github.com/cockroachdb/cockroach/client"
 	"github.com/cockroachdb/cockroach/storage"
+	"github.com/cockroachdb/cockroach/util"
 	"github.com/cockroachdb/cockroach/util/leaktest"
 )
 
@@ -69,21 +70,28 @@ func TestRaftLogQueue(t *testing.T) {
 		}
 	}
 
-	// Force a truncation check.
-	for _, store := range mtc.stores {
-		store.ForceRaftLogScanAndProcess(t)
-	}
+	// Sadly, occasionally the queue has a race with the force processing so
+	// this succeeds within will captures those rare cases.
+	var afterTruncationIndex uint64
+	util.SucceedsWithin(t, time.Second, func() error {
+		// Force a truncation check.
+		for _, store := range mtc.stores {
+			store.ForceRaftLogScanAndProcess(t)
+		}
 
-	// Ensure that firstIndex has increased indicating that the log
-	// truncation has occurred.
-	afterTruncationIndex, err := raftLeaderRepl.FirstIndex()
-	if err != nil {
-		t.Fatal(err)
-	}
-	if afterTruncationIndex <= originalIndex {
-		t.Fatalf("raft log has not been truncated yet, afterTruncationIndex:%d originalIndex:%d",
-			afterTruncationIndex, originalIndex)
-	}
+		// Ensure that firstIndex has increased indicating that the log
+		// truncation has occurred.
+		var err error
+		afterTruncationIndex, err = raftLeaderRepl.FirstIndex()
+		if err != nil {
+			t.Fatal(err)
+		}
+		if afterTruncationIndex <= originalIndex {
+			t.Fatalf("raft log has not been truncated yet, afterTruncationIndex:%d originalIndex:%d",
+				afterTruncationIndex, originalIndex)
+		}
+		return nil
+	})
 
 	// Force a truncation check again to ensure that attempting to truncate an
 	// already truncated log has no effect.

--- a/storage/client_test.go
+++ b/storage/client_test.go
@@ -235,6 +235,14 @@ func (m *multiTestContext) Start(t *testing.T, numStores int) {
 		m.transportStopper = stop.NewStopper()
 	}
 	m.transportStopper.AddCloser(m.transport)
+
+	// Wait for gossip to startup.
+	util.SucceedsWithin(t, 100*time.Millisecond, func() error {
+		if m.gossip.GetSystemConfig() == nil {
+			return util.Errorf("system config not available")
+		}
+		return nil
+	})
 }
 
 func (m *multiTestContext) Stop() {

